ark/utils/spatial_analysis_utils.py

```diff
@@ -9,30 +9,55 @@
 import os
 
 
-def calc_dist_matrix(label_map, path=None):
+def calc_dist_matrix(label_map, centroid_labels=None, path=None):
     """Generate matrix of distances between center of pairs of cells
 
     Args:
         label_map (np array): array with unique cells given unique pixel labels
+        centroid_labels (dict): the labels for each fov which are
+            needed to add correct coordinates to access distance matrix
+            needs to be of the same length and indexed corresponding to label_map.coords['fovs']
+            if None, then default assume that cell labels are in order for each fov
         path (string): path to save file. If None, then will directly return
     Returns:
         dist_matrix (dict): contains a cells x cells matrix with the euclidian
-            distance between centers of corresponding cells for every fov"""
+            distance between centers of corresponding cells for every fov, note that each distance matrix
+            is of type xarray"""
     # Check that file path exists, if given
+
     if path is not None:
         if not os.path.exists(path):
             raise ValueError("File path not valid")
+
     dist_mats_list = []
+
     # Extract list of fovs
     fovs = list(label_map.coords['fovs'].values)
-    for i in range(0, label_map.shape[0]):
-        props = skimage.measure.regionprops(label_map.loc[fovs[i], :, :, "segmentation_label"].values)
-        a = [props[j].centroid for j in range(len(props))]
-        centroids = np.array(a)
+
+    # generate centroid labels if None
+    if centroid_labels is None:
+        centroid_labels = {}
+
+        for fov_val in fovs:
+            fov_arr = label_map.loc[fov_val, :, :, 'segmentation_label'].values
+            fov_arr_labels = np.unique(fov_arr[fov_arr > 1]).tolist()
+            centroid_labels[fov_val] = fov_arr_labels
+
+    for i in range(len(fovs)):
+        # extract region properties of label map, then just get centroids
+        props = skimage.measure.regionprops(label_map.loc[fovs[i], :, :, 'segmentation_label'].values)
+        centroids = np.array([props[j].centroid for j in range(len(props))])
+
+        # generate the distance matrix, then assign centroid_labels as coords
         dist_matrix = cdist(centroids, centroids)
-        dist_mats_list.append(dist_matrix)
+        dist_mat_xarr = xr.DataArray(dist_matrix, coords=[centroid_labels[fovs[i]], centroid_labels[fovs[i]]])
+
+        # append final result to dist_mats_list
+        dist_mats_list.append(dist_mat_xarr)
+
     # Create dictionary to store distance matrices per fov
     dist_matrices = dict(zip([str(i) for i in fovs], dist_mats_list))
```

Comment:
are the members of `fovs` not strings?  if not, can we just have:
```fovs = list([str(fov) for fov in label_map.coords['fovs'].values])```
on line 35?

Reply:
So the problem is that sometimes, the members of fovs are ints. For example, in Erin's data. Can we safely assume that we'll only be getting strings or should we handle int fovs as well?

Reply:
The FOVs are ints in Erin's data because that's what Jay decided to name them. I think in general we should make sure FOVs are strings to avoid unexpected behavior. You can modify the underlying data to be Point7, Point8 instead of 7 and 
 8